ChatApplication/Services/CacheExtensionsService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Abp.Json;
+using ChatApplication.Data;
+using ChatApplication.Models;
+using ChatApplication.Services.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Distributed;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ChatApplication.Services
+{
+    public class CacheExtensionsService : ICacheExtensionsService
+    {
+        IDistributedCache _distributedCache;
+
+        public CacheExtensionsService(IDistributedCache distributedCache)
+        {
+            _distributedCache = distributedCache;
+        }
+
+        public async Task<IEnumerable<Message>> GetCacheValueAsync( string key)
+        {
+            var result = await _distributedCache.GetStringAsync(key);
+
+            if (String.IsNullOrEmpty(result))
+            {
+                return null;
+            }
+
+            var messageCache = JsonConvert.DeserializeObject<List<Message>>(result);
```

Comment:
Why are you deserializing to List<Message>? what if we want to store other objects in redis?